src/lib/rules/apple-touch-icons/apple-touch-icons.ts

```diff
@@ -0,0 +1,268 @@
+
+                return relValues.includes('apple-touch-icon') || relValues.includes('apple-touch-icon-precomposed');
+            });
+        };
+
+        const checkImage = async (appleTouchIcon: IAsyncHTMLElement, resource: string) => {
+            const appleTouchIconHref = normalizeString(appleTouchIcon.getAttribute('href'));
+
+            // Check if `href` doesn't exist, or it has the
+            // value of empty string.
+
+            if (!appleTouchIconHref) {
+                await context.report(resource, appleTouchIcon, `'apple-touch-icon' should have non-empty 'href' attribute`);
+
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // The following checks don't make sense for non-HTTP(S).
+
+            if (!isRegularProtocol(resource)) {
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            let appleTouchIconURL = '';
+
+            // If `href` exists and is not an empty string, try
+            // to figure out the full URL of the `apple-touch-icon`.
+
+            if (url.parse(appleTouchIconHref).protocol) {
+                appleTouchIconURL = appleTouchIconHref;
+            } else {
+                appleTouchIconURL = url.resolve(resource, appleTouchIconHref);
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            let networkData: INetworkData;
+
+            // Try to see if the `apple-touch-icon` file actually
+            // exists and is accesible.
+
+            try {
+                networkData = await context.fetchContent(appleTouchIconURL);
+            } catch (e) {
+                debug(`Failed to fetch the ${appleTouchIconHref} file`);
+                await context.report(resource, appleTouchIcon, `'${appleTouchIconHref}' file request failed`);
+
+                return;
+            }
+
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                await context.report(resource, appleTouchIcon, `'${appleTouchIconHref}' could not be fetched (status code: ${response.statusCode})`);
+
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Note: Async version of `image-size` doesn't work if the
+            // input is a Buffer.
+            //
+            // https://github.com/image-size/image-size/tree/4c527ba608d742fbb29f6d9b3c77b831b069cbb2#asynchronous
+
+            const image = getImageData(response.body.rawContent);
+
+            // Check if the image is a PNG.
+
+            if (image.type !== 'png') {
+                await context.report(resource, appleTouchIcon, `'${appleTouchIconHref}' is not a PNG`);
+            }
+
+            // Check if the image is 180x180px.
+
+            if (image.width !== 180 && image.height !== 180) {
```

Comment:
I think this should be `image.width !== 180 || image.height !== 180` just in case someone use an image with size 180x160 for example.

Reply:
Fixed and added a test, thanks!